theano/compile/function_module.py

```diff
@@ -752,9 +753,33 @@ def checkSV(sv_ori, sv_rpl):
         return f_cpy
 
     def __call__(self, *args, **kwargs):
+        """
+        Evaluates value of a function on given arguments.
+
+        Parameters
+        ----------
+        args : list
+            List of inputs to the function. All inputs are required, even when
+            some of them are not necessary to calculate requested subset of
+            outputs.
+
+        kwargs : dict
+            TODO: other kwargs?
+            Keyword argument `output_subset` is a list of indices of the
+            function's outputs that are requested to be calculated.
```

Comment:
Theano can have inputs with name and we can pass those parameter as kwargs. This isn't frequently used. We could just prevent a name as output_subset, or force people to use the apply interface for that. I'm not sure what is the best.

@lamblin @abergeron What do you think? We want to use a optional kwargs, but this would conflict with a named input of the same name that we would like to pass as a named parameter.